### PR TITLE
backend: (riscv) add vfadd and vfmul instructions

### PR DESCRIPTION
**Pull request by superlopuh**
- [x] make a note that the vfadd and vfmul with float registers is non-standard.

### CODE REVIEW

File: xdsl/dialects/riscv.py [hunk 1]
```diff
@@ -3272,6 +3272,41 @@ def assembly_line(self) -> str | None:
         )
 
 
+# endregion
+
+# region 17 "V" Standard Extension for Vector Operations
+
+# https://riscv.org/wp-content/uploads/2018/05/15.20-15.55-18.05.06.VEXT-bcn-v1.pdf
```
Comment (adutilleul) [inline xdsl/dialects/riscv.py:3279]: As far as I know the vector extension used in Snitch is non-standard (https://iis-git.ee.ethz.ch/smach/smallFloat-spec). Is this PR related to the paper at all, or are you also adding the standard vector extension for other purposes?

Comment (superlopuh) [reply xdsl/dialects/riscv.py:3279]: Ah yes this is purely for the paper, you make a good point, I'll note that.

Comment (nazavode) [inline xdsl/dialects/riscv.py:3279]:
```suggestion
# region 17 Xfvec packed SIMD non-standard extension, part of the smallFloat non-standard ISA extension set

# All Xfvec instructions performing vectorial single precision operations require 64bit floating point registers (a.k.a.: FLEN==64).
# https://iis-git.ee.ethz.ch/smach/smallFloat-spec/-/raw/master/smallFloat_isa.pdf
```

File: xdsl/dialects/riscv.py [hunk 2]
```diff
@@ -3272,6 +3272,41 @@ def assembly_line(self) -> str | None:
...
+
+@irdl_op_definition
+class VFAddSOp(
+    RdRsRsOperation[FloatRegisterType, FloatRegisterType, FloatRegisterType]
+):
+    """
+    Perform a pointwise single-precision floating-point addition over vectors.
+
+    On Snitch (and maybe other processors?) when the registers are floating point
+    registers on a 64-bit machine it treats them as registers containing a 2xf32 vector.
```
Comment (nazavode) [inline xdsl/dialects/riscv.py:3290]:
```suggestion
    Perform a vectorial single-precision floating-point addition on 2x single precision lanes packed in 64 bit floating point register operands.
```

File: xdsl/dialects/riscv.py [hunk 3]
```diff
@@ -3272,6 +3272,41 @@ def assembly_line(self) -> str | None:
...
+
+@irdl_op_definition
+class VFMulSOp(
+    RdRsRsOperation[FloatRegisterType, FloatRegisterType, FloatRegisterType]
+):
+    """
+    Perform a pointwise single-precision floating-point multiplication over vectors.
+
+    On Snitch (and maybe other processors?) when the registers are floating point
+    registers on a 64-bit machine it treats them as registers containing a 2xf32 vector.
```
Comment (nazavode) [inline xdsl/dialects/riscv.py:3304]:
```suggestion
   Perform a vectorial single-precision floating-point multiplication on 2x single precision lanes packed in 64 bit floating point register operands.
```

**Review by nazavode — APPROVED** [review]

**Review by compor — APPROVED** [review]

**Pull request merged**